src/libAtomVM/defaultatoms.c

```diff
@@ -103,6 +103,18 @@ static const char *const timeout_value_atom = "\xD" "timeout_value";
 static const char *const machine_atom = "\x7" "machine";
 static const char *const avm_floatsize_atom = "\xD" "avm_floatsize";
 
+static const char *const append_atom = "\x6" "append";
+static const char *const private_append_atom = "\xD" "private_append";
```

Comment:
I counted 14 chars, so it should be "\xE".

Reply:
Obviously this isn't tested. I haven't encountered `private_append` in generated beam code but found it in BEAM source code.
Not sure if we should remove the value.
Meanwhile, I did fix the length byte.

Reply:
default atoms allows us to reference them as constants, that can be used with statements such as switch, therefore if it is not referenced in our code it can be safely discarded.

Reply:
In addition it does consume memory even if not referenced.  I would prefer that `defaultatoms.[ch]` have a very conservative set of declarations that have a high probability of being used.  `ok`, `error`, `undefined` all come to mind.  Maybe a few others.

If possible I would err on the side of instantiating the atom only when needed, instead of at load time, but that is my personal opinion.

Reply:
`private_append` was used in the code but we had no test. I knew about it, it was a shortcoming of this PR, and was just too lazy to build an example. But then @bettio 's sharp eyes found out that it wouldn't work because the atom was wrongly defined.

So please let me do this properly as it should have been done. Scanning all OTP beams, I found many modules where this is genereated and will create a test case accordingly.

Reply:
I will try to take a look to this